improver/lightning.py

```diff
@@ -204,3 +204,220 @@ def latitude_to_threshold(
         rescale(latitude, (50.0, 10), (midlatitude, tropics), clip=True),
         rescale(latitude, (-50.0, -10), (midlatitude, tropics), clip=True),
     )
+
+
+class LightningMultivariateProbability(PostProcessingPlugin):
+    """
+    The algorithm outputs the probability of at least one lightning strike within
+    20 km of the location in a three hour period. The probabilities are calculated
+    for each individual ensemble member, and then they are averaged for the ensemble
+    probability forecast.
+
+    Inputs:
+    CAPE (cape; calculated with water loading and latent heat of freezing estimated)
+    Lifted Index (liftidx)
+    Precipitable Water (pwat in mm)
+    CIN (cin)
+    3-hour Accumulated Precipitation (apcp in inches)
+    Outputs:
+    20 km lightning probability over the valid time of the accumulated precipitation
+
+    ---------------------------+------------+----------------------------------------
+
+    Regression equation when CAPE and APCP are greater than zero:
```

Comment:
Just to ask, is this algorithm available anywhere in a referenceable location? If so, it would be great to add the reference.

Reply:
This actually came up as a question from the BoM as well when I had a call with them earlier this week.  Unfortunately, there is no reference available as this came from internal documentation that we received from the USAF about the algorithms they run in house in their current post-processing system. This algorithm had been developed from some previous statistical analysis the USAF had conducted.